front/app/modules/commercial/customizable_homepage_banner/admin/CTASettings/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { Section, SubSectionTitle } from 'components/admin/Section';
+import { FormattedMessage } from 'utils/cl-intl';
+import React from 'react';
+import messages from '../messages';
+import {
+  TAppConfigurationSetting,
+  IAppConfigurationSettings,
+} from 'services/appConfiguration';
+import CTASignedOutSettings from './CTASignedOutSettings';
+import CTASignedInSettings from './CTASignedInSettings';
+import { CLErrors } from 'typings';
+
+interface Props {
+  latestAppConfigSettings:
+    | IAppConfigurationSettings
+    | Partial<IAppConfigurationSettings>;
+  handleOnChange: (
+    settingName: TAppConfigurationSetting
+  ) => (settingKey: string, settingValue: any) => void;
```

Comment:
But it can be object or string and they have nothing in common. The next time `number` can be added and we will just add `number` to this list. So, for me in this case it's just extra typing without any gain.

Reply:
You're right.